x-pack/plugins/maps/public/classes/styles/vector/components/legend/marker_size_legend.tsx

```diff
@@ -38,6 +42,29 @@ export class MarkerSizeLegend extends Component<Props, State> {
 
   componentDidUpdate() {
     this._loadLabel();
+
+    if (this.state.maxLabelWidth === undefined && Object.values(this._textRefs).length) {
+      const missingRefElements = Object.values(this._textRefs).some((ref) => {
+        return ref.current == null;
+      });
+      if (!missingRefElements) {
+        let maxLabelWidth = 0;
+        Object.values(this._textRefs).forEach((ref) => {
+          const width = ref.current.getBBox().width;
```

Comment:
@nreese thanks for rewriting the logic, I did not know this was possible :raised_hands: